src/rgw/rgw_lua_background.cc

```diff
@@ -0,0 +1,51 @@
+#include "rgw_lua_background.h"
+#include "rgw_lua.h"
+#include "rgw_lua_utils.h"
+#include "include/ceph_assert.h"
+#include <lua.hpp>
+
+namespace rgw::lua {
+
+void Background::shutdown(){
+  this->stop();
+  runner.join();
+}
+void Background::stop(){
+  stopped = true;
+}
+
+//(1) Loads the script from the object
+//(2) Executes the script
+//(3) Sleep (configurable)
+void Background::run() {
+  this->create_background_metatable(L);
+
+  while (!stopped) {
+
+    std::string tenant;
+    auto rc = rgw::lua::read_script(dpp, store, tenant, null_yield, rgw::lua::context::background, rgw_script);
+    if (rc == -ENOENT) {
+      // no script, nothing to do
+    } else if (rc < 0) {
+      ldpp_dout(dpp, 1) << "WARNING: failed to read background script. error " << rc << dendl;
+    } else {
+      try {
+        //execute the background lua script
+        if (luaL_dostring(L, rgw_script.c_str()) != LUA_OK) {
```

Comment:
You could pre-compile the script and only recompile if it changes. Do you need to refetch (read_script) every loop?

Reply:
We could avoid unnecessary re fetching, I think that it is also worth implementing in other contexts (e.g post/pre request).
Should we descope it? (and apply it for all of the contexts at once) @yuvalif

Reply:
the script is cached, so, there is no I/O when reading it.
we can probably consider in a followup PR

Reply:
@batrick could you please re-review?